express/scripts/gnav.js

```diff
@@ -20,6 +20,55 @@ import {
 // eslint-disable-next-line import/no-unresolved
 } from './scripts.js';
 
+async function checkRedirect(currentPathName, geoLookup) {
+  const pathNameSplit = currentPathName.split('/');
+  let redirect = null;
+  if (pathNameSplit && pathNameSplit[1] && pathNameSplit[1] !== geoLookup) {
+    let subPath = geoLookup.length > 0 ? `/${geoLookup}` : '';
+    let i = 1;
+    while (pathNameSplit[i] !== 'express' && i < pathNameSplit.length) {
+      i += 1;
+    }
+    for (; i < pathNameSplit.length; i += 1) {
+      subPath = `${subPath}/${pathNameSplit[i]}`;
+    }
+    redirect = `${window.location.protocol}//${window.location.host}${subPath}${window.location.search}`;
+  }
+  return redirect;
+}
+
+async function geoCheck(userGeo, userLocale, geoCheckForce) {
+  const isInternationalCookie = document.cookie.match(/^(.*;)?\s*international\s*=\s*[^;]+(.*)?$/);
+  let redirect = null;
+  // DON'T if international cookie is set, unless geocheck=force is passed explicitly
+  if (geoCheckForce || !isInternationalCookie) {
+    const resp = await fetch(`${window.location.protocol}//${window.location.host}/express/system/geo-map.json`);
+    const json = await (resp.json());
+    let geoLookup = null;
+    for (let i = 0; i < json.data.length && geoLookup == null; i += 1) {
+      if (json.data[i].usergeo === userGeo) {
+        if (json.data[i].userlocales && json.data[i].userlocales.length > 0) {
+          const redirectLocalPaths = json.data[i].redirectlocalpaths.split(',');
+          const userLanguage = userLocale.substr(0, userLocale.indexOf('-'));
+          const userExpectedPath = `${userGeo.toLowerCase()}_${userLanguage}`;
+          for (let j = 0; j < redirectLocalPaths.length && geoLookup == null; j += 1) {
+            if (redirectLocalPaths[j] === userExpectedPath) {
```

Comment:
You may want to trim user input here:
```suggestion
            if (redirectLocalPaths[j].trim() === userExpectedPath) {
```

Reply:
Nice catch